src/components/DataDimension/DataElementSelector.js

```diff
@@ -86,59 +123,63 @@ const DataElementSelector = ({
         dataType: DIMENSION_TYPE_DATA_ELEMENT,
-        subGroup: TOTALS,
+        group,
+        subGroup,
     })
-    const [options, setOptions] = useState([])
-    const [loading, setLoading] = useState(true)
-    const [page, setPage] = useState(0)
-    const [hasNextPage, setHasNextPage] = useState(true)
-    const debouncedSearchTerm = useDebounce(searchInput, 200)
 
-    if (selectedItems.length) {
-        // FIXME: temporarily removes lint errors
-        console.log(loading + setFilter())
-    }
-
-    const fetchItems = useCallback(
-        async (searchTerm) => {
-            setLoading(true)
+    const fetchData = async () => {
+        try {
             const result = await apiFetchOptions({
                 dataEngine,
                 nameProp: displayNameProp,
-                page,
-                filter,
-                searchTerm,
+                filter: filterRef.current,
+                searchTerm: searchTermRef.current,
+                page: pageRef.current,
             })
-            const newOptions = []
-            result.dimensionItems?.forEach((item) => {
-                newOptions.push({
+
+            // XXX also check for length?
+            if (result?.dimensionItems) {
+                const newOptions = result.dimensionItems.map((item) => ({
                     label: item.name,
                     value: item.id,
                     disabled: item.disabled,
                     type: item.dimensionItemType,
                     expression: item.expression,
-                })
-            })
-            setLoading(false)
-            setOptions((prevOptions) =>
-                page > 1 ? [...prevOptions, ...newOptions] : newOptions
-            )
+                }))
+
+                setOptions((prevOptions) =>
+                    pageRef.current > 1
+                        ? [...prevOptions, ...newOptions]
+                        : newOptions
+                )
+            }
+
+            hasNextPageRef.current = result?.nextPage ? true : false
+
+            // XXX the following cannot be in here I believe, if it's still needed it should be triggered somewhere else, perhaps in a useEffect that listens to options
+            // but it's not possible to call fetchData() within the effect
 
-            setHasNextPage(result.nextPage)
-            /*  The following handles a very specific edge-case where the user can select all items from a 
-            page and then reopen the modal. Usually Transfer triggers the onEndReached when the end of 
-            the page is reached (scrolling down) or if too few items are on the left side (e.g. selecting 
-            49 items from page 1, leaving only 1 item on the left side). However, due to the way Transfer 
-            works, if 0 items are available, more items are fetched, but all items are already selected 
+            /*  The following handles a very specific edge-case where the user can select all items from a
```

Comment:
I think this can be removed. The edge-case gets even more edge'y for calculations, as the odds of someone already having the 50 first items in their calculation is very low (1. you probably don't write such long calculations, 2. there's no "add all" button as you have in the `Transfer` component, which makes it even harder to do).
If we find this to be a valid use case during our testing we can always add it in later. So I'd say ignore/remove this for now.

Reply:
Removed.